### PR TITLE
feat(card): add responsive layout

### PR DESCRIPTION
**Pull request by im-klau**
## Name PR according scheme: type of changes (feat/fix/refactor etc.) + description
feat(card): add responsive layout

## Jira issue

JIRA issue: [FSU-91](https://klau.atlassian.net/browse/FSU-91)

## Describe your changes
This PR introduces the responsive Card's layout.

## Checklist before requesting a review

- [ ] Checked which branch you would merge.
- [ ] Checked if the code follows the style guidelines of this project (use eslint and prettier).
- [ ] Performed a self-review of code (bugs, clean code rules, etc).
- [ ] Checked if added variables have unique names.
- [ ] Checked if added code/function is not a duplicate. 
- [ ] Tests have been added if possible.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Review by paulgrym — COMMENTED** [review]
image
When I set `hideIllustrationOnMobile: true` there is no padding on top of the Card and when we set `mobileLayout: "left"` card content is on the edge od the Card - no padding

**Comment by im-klau** [comment]
> image When I set `hideIllustrationOnMobile: true` there is no padding on top of the Card and when we set `mobileLayout: "left"` card content is on the edge od the Card - no padding

Well, actually this is how it was designed. I guess we shouldn't use this version with background colors.

Zrzut ekranu 2023-02-17 o 15 47 07

We need it in order to build this section on a tablet viewport (meaning a component without padding)
Zrzut ekranu 2023-02-17 o 15 48 28

Maybe I could just remove the bottom padding as well 🤔 to keep it more consistent

**Review by paulgrym — APPROVED** [review]
Everything's fine :) Thanks for explaining why the padding are what they are 😁

File: packages/design-system/patterns/card/Card.jsx [hunk 1]
```diff
@@ -54,28 +62,35 @@ Card.propTypes = {
...
   href: null,
   illustrationName: null,
+  hideIllustrationOnMobile: false,
   imageAlt: null,
-  imageHeight: "auto",
+  imageHeight: null,
   imageSrc: null,
+  layout: "left",
+  mobileLayout: "center",
   layout: "center",
```
Comment (im-klau) [inline packages/design-system/patterns/card/Card.jsx:94]:
```suggestion
  layout: "left",
  mobileLayout: "center",
```

**Pull request merged**